### PR TITLE
Add documentation for --dapr-internal-grpc-port in Dapr CLI run command

### LINKED ISSUE #2706: Add documentation for --dapr-internal-grpc-port in Dapr CLI run command
**What content needs to be created or modified?**

A new flag `--dapr-internal-grpc-port` (same as Daprd), is being added to the Dapr CLI.

**Describe the solution you'd like**

Explain what it does and the default behavior.

**Where should the new material be placed?**

https://github.com/dapr/docs/blob/v1.8/daprdocs/content/en/reference/cli/dapr-run.md

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/cli/pull/1040

**Additional context**

### PR DESCRIPTION
**Pull request by shubham1172**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Add an entry in `dapr run` CLI reference + add shorthand flags for HTTP and gRPC ports.

## Issue reference

This PR will close: #2706

### CODE REVIEW

**Comment by shubham1172** [comment]
~My commit is signed but DCO is broken because I first used v1.9 as the base branch. Can someone with access please re-trigger the DCO check?~ Force push fixed it.

**Review by mukundansundar — APPROVED** [review]

**Comment by msfussell** [comment]
It is not useful to say "this is an internal gRPC port" without explaining the usefulness of this. And the code PR https://github.com/dapr/cli/pull/1040 says nothing about why this was done. When and why would anyone want to set this in the CLI? (it would be so much nicer if we could revert the code PR). Also this is already in v1.9 yes?

**Comment by shubham1172** [comment]
@msfussell dapr/cli#1040 was an extension of the original PR dapr/cli#876, usually Dapr CLI is at parity with all flags from Daprd, including this one. So the CLI docs come from the daprd CLI reference - https://docs.dapr.io/reference/arguments-annotations-overview/. 

https://github.com/dapr/dapr/blob/a64a8a97c59c0678bec4d754b21d837d25a38f4c/pkg/runtime/cli.go#LL55C44-L55C52

File: daprdocs/content/en/reference/cli/dapr-run.md [hunk 1]
```diff
@@ -30,8 +30,9 @@ dapr run [flags] [command]
...
-| `--dapr-grpc-port`             | `DAPR_GRPC_PORT`     | `50001`                                                                            | The gRPC port for Dapr to listen on                                                                  |
-| `--dapr-http-port`             | `DAPR_HTTP_PORT`     | `3500`                                                                             | The HTTP port for Dapr to listen on                                                                  |
+| `--dapr-grpc-port`, `-G`       | `DAPR_GRPC_PORT`     | `50001`                                                                            | The gRPC port for Dapr to listen on                                                                  |
+| `--dapr-internal-grpc-port`, `-I` |                      | `50002`                                                                            | The gRPC port for the Dapr Internal API to listen on                                                         |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/cli/dapr-run.md]:
```suggestion
| `--dapr-internal-grpc-port`, `-I` |                      | `50002`                                                                            | gRPC port for the Dapr internal API to listen on; used when parsing the value from a local DNS component                                                        |
```

Grabbed this from dapr/cli#876 ....does this work? @msfussell

Comment (shubham1172) [reply daprdocs/content/en/reference/cli/dapr-run.md]: In this case, we should also update https://docs.dapr.io/reference/arguments-annotations-overview/ IMO.

Comment (hhunter-ms) [reply daprdocs/content/en/reference/cli/dapr-run.md]: @msfussell thoughts?

Comment (hhunter-ms) [reply daprdocs/content/en/reference/cli/dapr-run.md]: @msfussell friendly bump

Comment (msfussell) [reply daprdocs/content/en/reference/cli/dapr-run.md]: Yes that link also needs to also be updated. And internal should be lower case.
```suggestion
| `--dapr-internal-grpc-port`, `-I` |                      | `50002`                                                                            | The gRPC port for the Dapr internal API to listen on; used when parsing the value from a local DNS component                                                         |
```
This "; used when parsing the value from a local DNS component" still does not tell me why I would use this. Why would I possible ever change this to and when you say DNS component, are you referring the Name Resolution component?

Comment (hhunter-ms) [reply daprdocs/content/en/reference/cli/dapr-run.md]: @msfussell @shubham1172 does this help clarify that?

```suggestion
| `--dapr-internal-grpc-port`, `-I` |                      | `50002`                                                                            | The gRPC port for the Dapr internal API to listen on. Set during development for apps experiencing temporary errors with service invocation failures due to mDNS caching or going through a firewall. Can be any value greater than 1024 and must be different for each app.              |
```

Comment (shubham1172) [reply daprdocs/content/en/reference/cli/dapr-run.md]: > "Set during development for apps experiencing temporary errors with service invocation failures due to mDNS caching"...

Comment (hhunter-ms) [reply daprdocs/content/en/reference/cli/dapr-run.md]: @shubham1172 updated my suggestion ^^

File: daprdocs/content/en/reference/cli/dapr-run.md [hunk 2]
```diff
@@ -30,8 +30,9 @@ dapr run [flags] [command]
...
-| `--dapr-grpc-port`             | `DAPR_GRPC_PORT`     | `50001`                                                                            | The gRPC port for Dapr to listen on                                                                  |
-| `--dapr-http-port`             | `DAPR_HTTP_PORT`     | `3500`                                                                             | The HTTP port for Dapr to listen on                                                                  |
+| `--dapr-grpc-port`, `-G`       | `DAPR_GRPC_PORT`     | `50001`                                                                            | The gRPC port for Dapr to listen on                                                                  |
+| `--dapr-internal-grpc-port`, `-I` |                      | `50002`                                                                            | The gRPC port for the Dapr internal API to listen on. Set during development for apps experiencing temporary errors with service invocation failures due to mDNS caching or going through a firewall. Can be any value greater than 1024 and must be different for each app.              |
```
Comment (shubham1172) [inline daprdocs/content/en/reference/cli/dapr-run.md]:
```suggestion
| `--dapr-internal-grpc-port`, `-I` |                      | `50002`                                                                            | The gRPC port for the Dapr internal API to listen on. Set during development for apps experiencing temporary errors with service invocation failures due to mDNS caching, or configuring Dapr sidecars behind firewall. Can be any value greater than 1024 and must be different for each app.              |
```

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**